nx/lib/nx/binary_backend.ex

```diff
@@ -1115,15 +1117,52 @@ defmodule Nx.BinaryBackend do
   def triangular_solve(
         %{type: output_type} = out,
         %{type: a_type, shape: {rows, rows}} = a,
-        %{type: b_type, shape: {rows}} = b,
+        %{type: b_type, shape: b_shape} = b,
         opts
-      ) do
+      )
+      when b_shape == {rows, rows} or b_shape == {rows} do
     a_data = to_binary(a)
     b_data = to_binary(b)
-    out_bin = B.Decomposition.ts(a_data, a_type, b_data, b_type, rows, output_type, opts)
+    out_bin = B.Decomposition.ts(a_data, a_type, b_data, b_type, b_shape, output_type, opts)
     from_binary(out, out_bin)
   end
 
+  @doc false
+  def solve(%T{type: output_type} = output_holder, %T{shape: {n, n}} = a, %T{shape: b_shape} = b)
```

Comment:
Let's also expose this on `Nx` as `solve`. :) And given it works on existing tensors, there is no need for a backend function as well!

Reply:
This has the same issue as invert, albeit in a milder way. There are some algorithms which can be more efficient for solving non-triangularized matrices.

If we're exposing `solve`, I'd rather expose it delegating the implementation to each backend. Are there any major downsides in doing this?